dissect/target/loaders/velociraptor.py

```diff
@@ -36,14 +36,22 @@ def find_fs_directories(path: Path) -> tuple[Optional[OperatingSystem], Optional
 
     # Windows
     volumes = set()
+    vss_volumes = set()
     for accessor in WINDOWS_ACCESSORS:
         accessor_root = fs_root.joinpath(accessor)
         if accessor_root.exists():
             # If the accessor directory exists, assume all the subdirectories are volumes
-            volumes.update(accessor_root.iterdir())
+            for volume in accessor_root.iterdir():
+                # https://github.com/Velocidex/velociraptor/blob/87368e7cc678144592a1614bb3bbd0a0f900ded9/accessors/ntfs/vss.go#L82
+                if "HarddiskVolumeShadowCopy" in volume.name:
+                    vss_volumes.add(volume)
+                else:
+                    volumes.add(volume)
 
     if volumes:
-        return OperatingSystem.WINDOWS, list(volumes)
+        # The volumes that represent drives (C, D) are mounted first,
+        # otherwise one of the volume shadow copies could be detected as the root filesystem which results in errors.
+        return OperatingSystem.WINDOWS, list(volumes) + list(vss_volumes)
```

Comment:
```suggestion
        return OperatingSystem.WINDOWS, list(volumes | vss_volumes)
```

Reply:
The `volumes` should be the first items in the list using the join operator `|` cause the test to fail, because the order of the list is random.

```python

>>> first = {"C", "D"}
>>> second = {"VSS1", "VSS2"}

>>> list(first | second)
# Incorrect because the items of `second` should be added at the end of the list
['D', 'VSS1', 'C', 'VSS2'] 

>>> list(first) + list(second)
# Correct 
['D', 'C', 'VSS2', 'VSS1'] 
```